### PR TITLE
Fix storage clean up tool

### PR DESCRIPTION
**Pull request by zhangayqian**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [ ] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangayqian** [comment]
#### Before cleanup
image

#### cleanup
image

#### After clean up
image

File: server-base/src/main/java/org/apache/kylin/rest/job/StorageCleanupJob.java [hunk 1]
```diff
@@ -83,38 +86,27 @@ protected void execute(OptionsHelper optionsHelper) throws Exception {
...
         if (fs.exists(metadataPath)) {
-            FileStatus[] projectStatus = fs.listStatus(metadataPath);
-            if (projectStatus != null) {
-                for (FileStatus status : projectStatus) {
-                    String projectName = status.getPath().getName();
-                    if (!projects.contains(projectName)) {
-                        if (delete) {
-                            logger.info("Deleting HDFS path " + status.getPath());
-                            fs.delete(status.getPath(), true);
+            FileStatus[] fileStatus = fs.listStatus(metadataPath);
```
Comment (zzcclp) [inline server-base/src/main/java/org/apache/kylin/rest/job/StorageCleanupJob.java:95]: use fs.listStatus(file, PathFilter) to filter out 'protectedDir'

**Review by hit-lacus — APPROVED** [review]
Approved

File: server-base/src/test/java/org/apache/kylin/rest/job/StorageCleanupJobTest.java [hunk 2]
```diff
@@ -18,6 +18,8 @@
 
 package org.apache.kylin.rest.job;
 
+import com.google.inject.internal.cglib.proxy.$CallbackFilter;
+import org.apache.hadoop.fs.PathFilter;
```
Comment (zzcclp) [inline server-base/src/test/java/org/apache/kylin/rest/job/StorageCleanupJobTest.java:22]: remove these two imports.

**Pull request merged**